libs/heattracesidesheet/src/lib/utils-sidesheet/useGetEleNetwork.ts

```diff
@@ -0,0 +1,43 @@
+import { ElectricalNetwork, useContextId, useHttpClient } from '@cc-components/shared';
+import { useQuery } from '@tanstack/react-query';
+import { HeatTrace } from '@cc-components/heattraceshared';
+
+export const useGetEleNetwork = (ht: HeatTrace | null) => {
+  const client = useHttpClient();
+  const contextId = useContextId();
+
+  if (!ht) {
+    throw new Error(`Failed to resolve the heat trace`);
+  }
+
+  const { data, isLoading, error } = useQuery<ElectricalNetwork | null>({
+    queryKey: [ht.heatTraceCableNo, ht.facility, ht.project],

```

Comment:
QueryKey should contain info about which resource this is, e.g "echo". The way it is now it could cause a conflict in the future